src/index.js

```diff
@@ -53,11 +55,15 @@ class DropdownTreeSelect extends Component {
       showDropdown: this.props.showDropdown || false,
       searchModeOn: false
     }
+    // This needs id generator from https://github.com/dowjones/react-dropdown-tree-select/pull/220
+    this.clientId = this.props.id
   }
 
-  createList = ({ data, simpleSelect, showPartiallySelected, hierarchical }) => {
-    this.treeManager = new TreeManager({ data, simpleSelect, showPartiallySelected, hierarchical })
-    return this.treeManager.tree
+  initNewProps = ({ data, simpleSelect, radioSelect, showPartiallySelected, hierarchical }) => {
+    this.treeManager = new TreeManager({ data, simpleSelect, radioSelect, showPartiallySelected, hierarchical })
+    const { tree } = this.treeManager
+    const tags = this.treeManager.getTags()
```

Comment:
The line above this popped an idea - if we implement a getter for `tags` in tree manager, then we can do:

```
  ...
  const { tree, tags } = this.treeManager
  ...
```

`getTags` is basically a result of years of programming in ES5. Don't have to make that change in this PR though. :)

Reply:
Made some basic solution to this for now